src/lib/operators/import.cpp

```diff
@@ -66,15 +70,21 @@ std::shared_ptr<const Table> Import::_on_execute() {
 
   Hyrise::get().storage_manager.add_table(_tablename, table);
 
-  // must match ImportNode::output_expressions
+  // If a table encoding is specified, encode the table accordingly.
+  if (_table_encoding) {
+    const auto chunk_encoding_spec =
+        ChunkEncodingSpec(table->column_count(), SegmentEncodingSpec(_table_encoding.value()));
+    ChunkEncoder::encode_all_chunks(Hyrise::get().storage_manager.get_table(_tablename), chunk_encoding_spec);
```

Comment:
Is there a reason to obtain the table ptr again? It's already in the `table` variable.

Reply:
`SegmentEncodingSpec{*_table_encoding}`